aws/resource_aws_s3_bucket.go

```diff
@@ -337,6 +337,7 @@ func resourceAwsS3Bucket() *schema.Resource {
 			"replication_configuration": {
 				Type:     schema.TypeList,
 				Optional: true,
+				Computed: true,
```

Comment:
Hey quick question -- what was the motivation for adding this?

Reply:
Good question. The reason was - failed test

Reply:
In this case, we don't want to change this and the test failure you mention is unfortunately a red herring. S3 eventual consistency is a beast (we get failures regularly in our daily acceptance testing 😬 ) so you're not alone. I believe setting `Computed: true` here basically means that we will ignore out of band changes to the replication configuration in AWS if its not defined in the Terraform configuration, which is a change in behavior for the resource.

Removing it and running the tests worked fine for me 👍 
```
make testacc TEST=./aws TESTARGS='-run=TestAccAWSS3Bucket_Replication'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSS3Bucket_Replication -timeout 120m
=== RUN   TestAccAWSS3Bucket_Replication
--- PASS: TestAccAWSS3Bucket_Replication (87.72s)
=== RUN   TestAccAWSS3Bucket_ReplicationWithoutStorageClass
--- PASS: TestAccAWSS3Bucket_ReplicationWithoutStorageClass (57.96s)
=== RUN   TestAccAWSS3Bucket_ReplicationExpectVersioningValidationError
--- PASS: TestAccAWSS3Bucket_ReplicationExpectVersioningValidationError (32.61s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	178.355s
```

Reply:
Thanks for the great explanation, now I revert it.
But for me the red flag was - that this test always pass before I made other changes. I explained it 2 step bucket creation process (first `Create` without replication, next `Update` and than we `Read` it several times while it come to state). On the first `Read` replication can absent, but what's changed - I added default line for `replication` to state and it can be read. 
While debug output shows that all is ok, looks like test read first version of state if the section is present